aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,428 @@
+				"min_vcpus":           computeResource.MinvCpus,
+				"security_group_ids":  schema.NewSet(schema.HashString, flattenStringList(computeResource.SecurityGroupIds)),
+				"spot_iam_fleet_role": computeResource.SpotIamFleetRole,
+				"subnets":             schema.NewSet(schema.HashString, flattenStringList(computeResource.Subnets)),
+				"tags":                tagsToMapGeneric(computeResource.Tags),
+				"type":                computeResource.Type,
+			},
+		})
+	}
+
+	d.Set("arn", computeEnvironment.ComputeEnvironmentArn)
+	d.Set("ecc_cluster_arn", computeEnvironment.ComputeEnvironmentArn)
+	d.Set("status", computeEnvironment.Status)
+	d.Set("status_reason", computeEnvironment.StatusReason)
+
+	return nil
+}
+
+func resourceAwsBatchComputeEnvironmentDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	log.Printf("[DEBUG] Delete compute environment \"%s\".\n", computeEnvironmentName)
+
+	updateInput := &batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(computeEnvironmentName),
+		State:              aws.String("DISABLED"),
+	}
+
+	if _, err := conn.UpdateComputeEnvironment(updateInput); err != nil {
+		return err
+	}
+
+	if err := waitComputeEnvironmentValid(d, meta); err != nil {
+		return err
+	}
+
+	input := &batch.DeleteComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(computeEnvironmentName),
+	}
+
+	if _, err := conn.DeleteComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	if err := waitComputeEnvironmentDeleted(d, meta); err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceAwsBatchComputeEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	log.Printf("[DEBUG] Update compute environment \"%s\".\n", computeEnvironmentName)
+
+	input := &batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(computeEnvironmentName),
+		ComputeResources:   &batch.ComputeResourceUpdate{},
+	}
+
+	if d.HasChange("service_role") {
+		log.Printf("[DEBUG] Service role of \"%s\" has change.\n", computeEnvironmentName)
+		input.ServiceRole = aws.String(d.Get("service_role").(string))
+	}
+	if d.HasChange("state") {
+		log.Printf("[DEBUG] State of \"%s\" has change.\n", computeEnvironmentName)
+		input.ServiceRole = aws.String(d.Get("state").(string))
+	}
+
+	if d.HasChange("compute_resources") {
+		log.Printf("[DEBUG] Compute resources of \"%s\" has change.\n", computeEnvironmentName)
+		computeResources := d.Get("compute_resources").([]interface{})
+		if len(computeResources) == 0 {
+			return fmt.Errorf("One compute environment is expected, but no compute environments are set")
+		}
+		computeResource := computeResources[0].(map[string]interface{})
+
+		if d.HasChange("compute_resources.0.desired_vcpus") {
```

Comment:
Looking at the HasChange code, I'm not sure we even need to check for children: we could simply pass options, since we know that `compute_resources ` changed. Thougths?